src/hb-subset-input.cc

```diff
@@ -269,98 +313,153 @@ hb_subset_input_get_retain_all_features (hb_subset_input_t *subset_input)
+ **/
+HB_EXTERN hb_set_t *
+hb_subset_input_no_subset_tables_set (hb_subset_input_t *subset_input)
 {
-  subset_input->drop_hints = drop_hints;
+  return subset_input->no_subset_tables;
 }
 
-HB_EXTERN hb_bool_t
-hb_subset_input_get_drop_hints (hb_subset_input_t *subset_input)
-{
-  return subset_input->drop_hints;
-}
-
-HB_EXTERN void
-hb_subset_input_set_desubroutinize (hb_subset_input_t *subset_input,
-				    hb_bool_t desubroutinize)
-{
-  subset_input->desubroutinize = desubroutinize;
-}
 
+/**
+ * hb_subset_input_get_flag:
+ * @subset_input: a subset_input.
+ * @flag: which flag to check.
+ *
+ * Return value: value of the specified flag.
+ *
+ * Since: REPLACE
+ **/
 HB_EXTERN hb_bool_t
-hb_subset_input_get_desubroutinize (hb_subset_input_t *subset_input)
+hb_subset_input_get_flag (hb_subset_input_t *input,
+			  hb_subset_flag_t flag)
 {
-  return subset_input->desubroutinize;
+  switch (flag)
+  {
+    case HB_SUBSET_FLAG_HINTING:
+      return !input->drop_hints;
+    case HB_SUBSET_FLAG_RETAIN_GIDS:
+      return input->retain_gids;
+    case HB_SUBSET_FLAG_DESUBROUTINIZE:
+      return input->desubroutinize;
+    case HB_SUBSET_FLAG_NAME_LEGACY:
+      return input->name_legacy;
+    case HB_SUBSET_FLAG_SET_OVERLAPS_FLAG:
+      return input->overlaps_flag;
+    case HB_SUBSET_FLAG_PASSTHROUGH_UNRECOGNIZED:
+      return input->passthrough_unrecognized;
+    case HB_SUBSET_FLAG_NOTDEF_OUTLINE:
+      return input->notdef_outline;
+    default:
+      return false;
+  }
 }
 
 /**
- * hb_subset_input_set_retain_gids:
+ * hb_subset_input_set_flag:
  * @subset_input: a subset_input.
- * @retain_gids: If true the subsetter will not renumber glyph ids.
- * Since: 2.4.0
+ * @flag: which flag to set.
+ * @value: new value for the flag.
+ *
+ * Set the specified flag to @value.
+ *
+ * Since: REPLACE
  **/
 HB_EXTERN void
-hb_subset_input_set_retain_gids (hb_subset_input_t *subset_input,
-				 hb_bool_t retain_gids)
+hb_subset_input_set_flag (hb_subset_input_t *input,
+			  hb_subset_flag_t flag,
+			  hb_bool_t value)
 {
-  subset_input->retain_gids = retain_gids;
+  switch (flag)
+  {
+    case HB_SUBSET_FLAG_HINTING:
+      input->drop_hints = !value;
+      break;
+    case HB_SUBSET_FLAG_RETAIN_GIDS:
+      input->retain_gids = value;
+      break;
+    case HB_SUBSET_FLAG_DESUBROUTINIZE:
+      input->desubroutinize = value;
+      break;
+    case HB_SUBSET_FLAG_NAME_LEGACY:
+      input->name_legacy = value;
+      break;
+    case HB_SUBSET_FLAG_SET_OVERLAPS_FLAG:
+      input->overlaps_flag = value;
+      break;
+    case HB_SUBSET_FLAG_PASSTHROUGH_UNRECOGNIZED:
+      input->passthrough_unrecognized = value;
+      break;
+    case HB_SUBSET_FLAG_NOTDEF_OUTLINE:
+      input->notdef_outline = value;
+    default:
+      // Do nothing.
+      break;
+  }
 }
 
 /**
- * hb_subset_input_get_retain_gids:
- * Returns: value of retain_gids.
- * Since: 2.4.0
+ * hb_subset_input_set_user_data: (skip)
+ * @face: A subset input object
```

Comment:
This should be `@input`. There probably should be some consistency in naming this variable, either input or subset_input.

Reply:
Alright, I standardized everything to use "input".